### PR TITLE
added support for application id 3

### PR DESCRIPTION
**Pull request by jaroszan**
I have added two lines in base dictionary:

    `<application id="3"> '
    '</application>`

Run autogen.sh script and it seems to work. At least client does not complain about acct-application-id 3 anymore. Unfortunately I do not have real cases to test it.

### CODE REVIEW

**Comment by fiorix** [comment]
Test still failing?

File: diam/dict/util_test.go [hunk 1]
```diff
@@ -18,9 +18,17 @@ func TestApps(t *testing.T) {
 	if apps[0].ID != 0 {
 		t.Fatalf("Unexpected app.ID. Want 0, have %d", apps[0].ID)
 	}
+	// Base accounting.
+	if apps[1].ID != 3 {
+		t.Fatalf("Unexpected app.ID. Want 0, have %d", apps[1].ID)
```
Comment (fiorix) [inline diam/dict/util_test.go]: fix: Want 3, have %d

**Comment by fiorix** [comment]
A couple more comments to fix tests error messages.

**Comment by jaroszan** [comment]
I wonder whether the following modification makes more sense (https://github.com/fiorix/go-diameter/blob/master/diam/sm/smparser/app.go), from:

`if id == 0xffffffff`

to:

`if id == 0xffffffff || id == 3`

**Comment by fiorix** [comment]
Hmmm not sure either. Would have to revisit the RFC, can't recall this.

**Comment by jaroszan** [comment]
This one is a subset of https://github.com/fiorix/go-diameter/pull/56 , closing

**Pull request closed without merging**